docs/realtime/Protocol.md

```diff
@@ -0,0 +1,323 @@
+# RTC protocol
+
+The WebSocket API provides means to receive events on change resources (that the JWT has access to).
+
+## Connecting to the WebSocket API
+
+Build the URL according to the template below:
+
+```
+wss://<project-id>.<zone>.app.jexia.com/rtc?access_token=<jwt>
+```
+ARGUMENT     | TYPE   | DESCRIPTION                           | EXAMPLE
+-------------|--------|---------------------------------------|----------------------------------------
+project-id   | UUID   | unique identifier/hash of the project | `4f8c6ee3-f2d1-47b8-83c4-c5250b86cab4`
+zone         | string | project zone                          | `nl00`
+jwt          | string | valid token obtained after logging into the project using the API key credentials or any other method | `eyJhbGciOiJS... ...aev96IlwB1OFzDmyw`
+
+Connecting to the WebSocket API is done using a WebSocket client and connect it to the WebSocket address. Use one of the available WebSocket clients to create a connection:
+- [Simple WebSocket Client](https://chrome.google.com/webstore/detail/simple-websocket-client/pfdhoblngboilpfeibdedpjgfnlcodoo) (Chrome extension)
+- [Smart Websocket Client](https://chrome.google.com/webstore/detail/smart-websocket-client/omalebghpgejjiaoknljcfmglgbpocdp) (Chrome extension)
+- [WebSocket King](https://websocketking.com) (Online WebSocket client)
+
+## Before you start
+1. Create a User/API key.
+2. Log In providing valid credentials and method (check [here](../auth/README.md) for documentaion).
+3. Create some resources with WebApp or management API.
+4. Make sure that User/API key you are going to use has all reuired permissions (policies) on the resources, create policies if not.
+
+## Sending and Receiving messages
+After establishing the connection to the WebSocket API, the client can send messages to perform actions, like subscribing to particular events and receive messages, like notifications about changes. Both send and received messages must be using JSON encoding.
+
+```json
+{
+  "type": "<message type>",
+  "data": "<message data>"
+}
+```
+
+ARGUMENT | TYPE   | DESCRIPTION            | EXAMPLE
+---------|--------|------------------------|-----------------------------------------------
+type     | String | message type           | `subscribe` / `unsubscribe` / `publish` ...
+data     | Any    | message data (payload) | `{"command":"unsubscribe","arguments":{...}}`
+
+
+### Command messages
+
+```json
+{
+  "type": "command",
+  "data": {
+    "command": "<command>",
+    "correlation_id": "<correlation ID: any valid JSON value>",
+    "arguments": { "<command arguments>" }
+  }
+}
+```
+
+Each command has the `command` field in the `data` object to specify the command and optionally an `arguments` object to specify the arguments of the command.
+Field `correlation_id` is OPTIONAL and ignored by RTC, it is generated/provided by the client and returned back as apart of command response AS IS (in case it was provided by WebSocket client).
```

Comment:
```suggestion
Field `correlation_id` is OPTIONAL and ignored by RTC, it is generated/provided by the client and returned back as a part of the command response AS IS (in case it was provided by WebSocket client).
```